src/data/migration.ts

```diff
@@ -0,0 +1,191 @@
+import { ColumnHint, Filter, QueryBuilderOptions, QueryType, SelectedColumn } from "types/queryBuilder";
+import { CHBuilderQuery, CHQuery, CHSqlQuery, EditorType } from "types/sql";
+import { isVersionGtOrEq, pluginVersion } from "utils/version";
+import { mapGrafanaFormatToQueryType } from "./utils";
+
+export type AnyCHQuery = Partial<CHQuery> & {[k: string]: any};
+export type AnyQueryBuilderOptions = Partial<QueryBuilderOptions> & {[k: string]: any};
+
+/**
+ * Takes a CHQuery and transforms it to the latest interface.
+ * Returns undefined if no migration is applied.
+ */
+export const migrateCHQuery = (savedQuery: CHQuery): CHQuery | undefined => {
+  if (savedQuery.rawSql === undefined) {
+    return undefined;
```

Comment:
Why not return the same `savedQuery` to have a predictable signature `(CHQuery) => CHQuery`?